civis/ml/_model.py

```diff
@@ -493,13 +493,27 @@ def train_data(self):
     def table(self):
         self.result()  # Block and raise errors if any
         if self._table is None:
+            # An index column will only be present if primary key is
+            # metadata path to input parameters is different
+            # for training and prediction
+            if self.is_training:
+                pkey = self.metadata[
+                    'run']['configuration']['data']['primary_key']
+            else:
+                pkey = self.metadata[
+                    'jobs'][0]['run']['configuration']['data']['primary_key']
```

Comment:
Do you think that it's worth pulling this logic block out into a private `_table_primary_key` (or something like that) method? Parsing out the primary key's column name feels logically distinct from this method, which is downloading the table produced by the CivisML job.

Reply:
Yeah, I was on the fence about that myself. I'll pull it into a helper method.